### PR TITLE
Added service ID validation and modified version ID validation.

### PR DESCRIPTION
**Pull request by Honcharov12**
Now service ID and version ID are validating according to GAE service and version IDs.

### CODE REVIEW

**Comment by Honcharov12** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/5597/

File: AdminServer/appscale/admin/__init__.py [hunk 1]
```diff
@@ -507,10 +507,16 @@ def delete(self, project_id, service_id):
 class VersionsHandler(BaseHandler):
   """ Manages service versions. """
 
-  # A rule for validating version IDs.
-  VERSION_ID_RE = re.compile(r'(?!-)[a-z\d\-]{1,100}')
+  # A rule for validating project IDs.
   PROJECT_ID_RE = re.compile(r'^[a-z][a-z\d\-]{5,29}$')
 
+  # A rule for validating version IDs.
+  VERSION_ID_RE = re.compile(r'^(?:^(?!-)[a-z\d\-]{0,62}[a-z\d]$)$')
```
Comment (scragraham) [inline AdminServer/appscale/admin/__init__.py]: I think replacing \d with 0-9 makes the regular expression a little more readable. Also you shouldn't have to include the re in (?:) Also when '-' is used at the end of the set of characters you don't need to escape it
```suggestion
  VERSION_ID_RE = re.compile(r'^(?!-)[a-z0-9-]{0,62}[a-z0-9]$')
```

File: AdminServer/appscale/admin/__init__.py [hunk 2]
```diff
@@ -507,10 +507,16 @@ def delete(self, project_id, service_id):
...
+
+  # A rule for validating service IDs.
+  SERVICE_ID_RE = re.compile(r'^(?:^(?!-)[a-z\d\-]{0,62}[a-z\d]$)$')
```
Comment (scragraham) [inline AdminServer/appscale/admin/__init__.py]:
```suggestion
  SERVICE_ID_RE = re.compile(r'^(?!-)[a-z0-9-]{0,62}[a-z0-9]$')
```

File: AdminServer/appscale/admin/__init__.py (hunk 1 above)
Comment (Honcharov12) [reply AdminServer/appscale/admin/__init__.py]: Thank you, Steven! This re was suggested by gcloud, but I also find it a little bit tricky :)

**Comment by Honcharov12** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/5604/

**Comment by Honcharov12** [comment]
**Demo**

1. Service ID.

Earlier, if user had specified invalid service ID, he received next type of error:
```
Deploying service qqA for guestbook
Invalid response: <html><title>404: Not Found</title><body>404: Not Found</body></html>
```

with following log:
```
stacktrace : Traceback (most recent call last):
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/scripts/appscale.py", line 90, in main
    appscale.deploy(sys.argv[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale.py", line 578, in deploy
    login_host, http_port = AppScaleTools.upload_app(options)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale_tools.py", line 1002, in upload_app
    operation_id = admin_client.create_version(version, remote_file_path)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 212, in call
    raise attempt.get()
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 247, in get 
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 114, in create_version
    operation = self.extract_response(response)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 60, in extract_response
    raise AdminError('Invalid response: {}'.format(response.content))
AdminError: Invalid response: <html><title>404: Not Found</title><body>404: Not Found</body></html>

exception : AdminError

tools_version : 3.6.0

platform : Linux-4.4.0-138-generic-x86_64-with-Ubuntu-16.04-xenial

message : Invalid response: <html><title>404: Not Found</title><body>404: Not Found</body></html>

runtime : CPython
```

Now it returns clearer error message:
```
Deploying service qqA for guestbook
Invalid service ID. May only contain lowercase letters, digits, and hyphens. Must begin and end with a letter or digit. Must not exceed 63 characters.
```

with following log:
```
stacktrace : Traceback (most recent call last):
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/scripts/appscale.py", line 90, in main
    appscale.deploy(sys.argv[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale.py", line 578, in deploy
    login_host, http_port = AppScaleTools.upload_app(options)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale_tools.py", line 1002, in upload_app
    operation_id = admin_client.create_version(version, remote_file_path)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 212, in call
    raise attempt.get()
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 247, in get 
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 114, in create_version
    operation = self.extract_response(response)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 69, in extract_response
    raise AdminError(message)
AdminError: Invalid service ID. May only contain lowercase letters, digits, and hyphens. Must begin and end with a letter or digit. Must not exceed 63 characters.

exception : AdminError

tools_version : 3.6.0

platform : Linux-4.4.0-138-generic-x86_64-with-Ubuntu-16.04-xenial

message : Invalid service ID. May only contain lowercase letters, digits, and hyphens. Must begin and end with a letter or digit. Must not exceed 63 characters.

runtime : CPython
```
==================================================================
2. Version ID
**Note**: taking into account, that in current version AppScale does not support application versions, it is impossible to reproduce such kind of error without modifying AppScale and AppScale tools. 
In AppScale we need to comment:
https://github.com/AppScale/appscale/blob/25331e2a7b6059a64944f045fd6c4639153b5822/AdminServer/appscale/admin/__init__.py#L582-L584
In AppScale tools we need to replace DEFAULT_VERSION with some kind of invalid version id, such as 'qqA':
https://github.com/AppScale/appscale-tools/blob/71b767226207a53a10b771a18b8f67149329e01f/appscale/tools/admin_api/client.py#L90

Earlier, it used to deploy application with such version ID.
Now, if changes applied only to AppScale tools it returns:
`AppScale currently does not support versions, so you have to use default version ID, i.e. "v1"`

with next log file:
```
stacktrace : Traceback (most recent call last):
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/scripts/appscale.py", line 90, in main
    appscale.deploy(sys.argv[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale.py", line 578, in deploy
    login_host, http_port = AppScaleTools.upload_app(options)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale_tools.py", line 1002, in upload_app
    operation_id = admin_client.create_version(version, remote_file_path)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 212, in call
    raise attempt.get()
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 247, in get 
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 114, in create_version
    operation = self.extract_response(response)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 69, in extract_response
    raise AdminError(message)
AdminError: AppScale currently does not support versions, so you have to use default version ID, i.e. "v1"

exception : AdminError

tools_version : 3.6.0

platform : Linux-4.4.0-138-generic-x86_64-with-Ubuntu-16.04-xenial

message : AppScale currently does not support versions, so you have to use default version ID, i.e. "v1"

runtime : CPython
```

And if both changes are applied, we get:
`Invalid version ID. May only contain lowercase letters, digits, and hyphens. Must begin and end with a letter or digit. Must not exceed 63 characters.`

with log file:
```
stacktrace : Traceback (most recent call last):
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/scripts/appscale.py", line 90, in main
    appscale.deploy(sys.argv[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale.py", line 578, in deploy
    login_host, http_port = AppScaleTools.upload_app(options)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/appscale_tools.py", line 1002, in upload_app
    operation_id = admin_client.create_version(version, remote_file_path)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 212, in call
    raise attempt.get()
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 247, in get 
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 114, in create_version
    operation = self.extract_response(response)
  File "/home/vovan/venvs/appscale-tools/local/lib/python2.7/site-packages/appscale/tools/admin_api/client.py", line 69, in extract_response
    raise AdminError(message)
AdminError: Invalid version ID. May only contain lowercase letters, digits, and hyphens. Must begin and end with a letter or digit. Must not exceed 63 characters.

exception : AdminError

tools_version : 3.6.0

platform : Linux-4.4.0-138-generic-x86_64-with-Ubuntu-16.04-xenial

message : Invalid version ID. May only contain lowercase letters, digits, and hyphens. Must begin and end with a letter or digit. Must not exceed 63 characters.

runtime : CPython
```

**Pull request merged**